Jenkinsfile

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env groovy
+
+// These are the platforms we are building against.
+def platforms = ["linux", "mac", "windows"]
+
+// This generates the launchConfig for a particular build.
+def generateConfig(deploymentName, templateUrl) {
+    return """
+---
+launch_config_version: 1
+deployment_name: ${deploymentName}
+template_url: ${templateUrl}
+provider: aws
+aws_region: us-west-1
+template_parameters:
+    KeyName: default
+    AdminLocation: 0.0.0.0/0
+    PublicSlaveInstanceCount: 1
+    SlaveInstanceCount: 1
+"""
+}
+
+
+// This class abstracts away the functions required to create a test cluster
+// for each of our platforms.
+class TestCluster implements Serializable {
+    WorkflowScript script
+    String platform
+    int createAttempts
+
+    TestCluster(WorkflowScript script, String platform) {
+        this.script = script
+        this.platform = platform
+        this.createAttempts = 0
+    }
+
+    def _create() {
+        script.sh "./dcos-launch create -c ${this.platform}_config.yaml -i ${this.platform}_cluster_info.json"
+    }
+
+    def destroy() {
+        script.sh "./dcos-launch delete -i ${this.platform}_cluster_info.json"
+    }
+
+    def _block() {
+        script.sh "./dcos-launch wait -i ${this.platform}_cluster_info.json"
+    }
+
+    def create() {
+        script.sh "rm -rf ${this.platform}_config.yaml"
+        script.sh "rm -rf ${this.platform}_cluster_info.json"
+
+        script.writeFile([
+            "file": "${this.platform}_config.yaml",
+            "text" : script.generateConfig(
+                "dcos-cli-${this.platform}-${script.env.BRANCH_NAME}-${script.env.BUILD_ID}-${this.createAttempts}",
+                "${script.env.CF_TEMPLATE_URL}")])
+
+        _create()
+
+        createAttempts++
+    }
+
+    def block() {
+        while (true) {
+            try {
+                _block()
+                break
+            } catch(Exception e) {
+                destroy()
+                if (e instanceof InterruptedException) {
+                    break
+                }
+                create()
+            }
+        }
+    }
+
+    def dcosUrl() {
+        script.sh """
+            ./dcos-launch describe -i ${this.platform}_cluster_info.json \
+            | python -c \
+                'import sys, json; \
+                 contents = json.load(sys.stdin); \
+                 print(contents["masters"][0]["public_ip"], end="")' \
+            > ${this.platform}_dcos_url"""
+
+        return script.readFile("${this.platform}_dcos_url")
+    }
+
+    def acsToken() {
+        def dcosUrl = this.dcosUrl()
+
+        script.sh """
+            python -c \
```

Comment:
Meaning a script file in the repo that we just call out to? I was hoping to avoid having individual script files like this just for CI, but I could be convinced otherwise.